backend/open_webui/routers/functions.py

```
@@ -42,6 +51,97 @@ async def get_functions(user=Depends(get_admin_user)):
+
+
+############################
+# SyncFunctions
+############################
+
+
+class SyncFunctionsForm(FunctionForm):
+    functions: list[FunctionModel] = []
+
+
+@router.post("/sync", response_model=Optional[FunctionModel])
```

Ruff (0.11.9)

Use `X | Y` for type annotations

(UP007)